pkg/encoding/v1/encoding.go

```diff
@@ -440,15 +501,28 @@ func (d *Decoder) Decode(data []byte) (*object.OpenCompose, error) {
 					mount.ReadOnly = *m.ReadOnly
 				}
 
+				if m.SecretRef != nil {
+					mount.SecretRef = m.SecretRef
+				}
+
 				oc.Mounts = append(oc.Mounts, mount)
 			}
 
 			// convert env
 			for _, e := range c.Env {
-				oc.Environment = append(oc.Environment, object.EnvVariable{
-					Key:   e.Key,
-					Value: e.Value,
-				})
+				env := object.EnvVariable{
+					Key: e.Key,
+				}
+
+				if e.Value != nil {
+					env.Value = e.Value
+				}
+
+				if e.SecretRef != nil {
```

Comment:
this should be `else if` of the above right? because only one field should be valid right at a time? either `Value` or `SecretRef`? And the `else` should be an error?